src/clean.cc

```diff
@@ -238,21 +238,21 @@ int Cleaner::CleanRule(const char* rule) {
   return status_;
 }
 
-int Cleaner::CleanRules(int rule_count, char* rules[]) {
+int Cleaner::CleanRules(const std::vector<std::string>& rules) {
```

Comment:
could this instead be a pair or iterators to vector?

Or a template function that takes generic iterators?